elftools/elf/elffile.py

```diff
@@ -588,12 +608,18 @@ def _identify_file(self):
     def _section_offset(self, n):
         """ Compute the offset of section #n in the file
         """
-        return self['e_shoff'] + n * self['e_shentsize']
+        shentsize = self['e_shentsize']
+        if self['e_shoff'] > 0 and shentsize < (0x40 if self.elfclass == 64 else 0x28):
```

Comment:
can we derive these values (0x40, 0x28, 0x38, 0x20) from something instead of being magic constants?

Reply:
The constants come straight from the spec, but I might be able can get them as parser sizeofs.